src/coreclr/jit/llvm.cpp

```diff
@@ -27,6 +27,9 @@ using llvm::Module;
 static Module*          _module    = nullptr;
 static llvm::DIBuilder* _diBuilder = nullptr;
 static LLVMContext _llvmContext;
+
+static Function* _nullCheckFunction = nullptr;
```

Comment:
Unprotected (with locks) static state like this is concerning. It is fine as-is currently, but we should make it more apparent what's missing.

Ideally, there'd be a `StaticCompiler`-like class (which should also encapsulate other `static` things above) that would be responsible for managing state shared across compilations. This would also make building SPMI-like tooling for the LLVM Jit easier.

Reply:
Thanks, added a comment.